### PR TITLE
Deep Q-Learning article

### PR DESCRIPTION
**Pull request by simoninithomas**
Here's Unit 3 Article 🥳 .

The whole article has been reviewed with Grammarly. It's just for small checkups. Thanks @echarlaix for your help 🤗

### CODE REVIEW

File: deep-rl-dqn.md [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+
+2. **Avoid forgetting previous experiences and reduce the correlation between experiences**.
+- The problem we get if we give sequential samples of experiences to our neural network is that our network tends to forget **the previous experiences as it overwrites new experiences.** For instance, if we are in the first level and then the second, which is different, our agent can forget how to behave and play in the first level.
+
+The solution is to create a Replay Buffer that stores experience tuples while interacting with the environment and then sample a small batch of tuples. This prevents **the network from only learning about what it has immediately done.**
+
+Experience replay also has other benefits. By randomly sampling the experiences, we remove correlation in the observation sequences and avoid **action values from oscillating or diverging catastrophically.**
+
+In the Deep Q-Learning pseudocode, we see that we **initialize a replay memory buffer D from capacity N** (N is an hyperparameter that you can define). We then,  store experiences in the memory and then sample a minibatch of experiences to feed the Deep Q-Network during the training phase.
```
Comment (echarlaix) [inline deep-rl-dqn.md:178]:
```suggestion
In the Deep Q-Learning pseudocode, we see that we **initialize a replay memory buffer D from capacity N** (N is an hyperparameter that you can define). We then, store experiences in the memory and then sample a minibatch of experiences to feed the Deep Q-Network during the training phase.
```
```suggestion
In the Deep Q-Learning pseudocode, we see that we **initialize a replay memory buffer D from capacity N** (N is an hyperparameter that you can define). We then,  store experiences in the memory and then sample a minibatch of experiences to feed the Deep Q-Network during the training phase.
```

Comment (echarlaix) [inline deep-rl-dqn.md:178]:
```suggestion
In the Deep Q-Learning pseudocode, we see that we **initialize a replay memory buffer D from capacity N** (N is an hyperparameter that you can define). We then, store experiences in the memory and then sample a minibatch of experiences to feed the Deep Q-Network during the training phase.
```
```suggestion
In the Deep Q-Learning pseudocode, we see that we **initialize a replay memory buffer D from capacity N** (N is an hyperparameter that you can define). We then,  store experiences in the memory and then sample a minibatch of experiences to feed the Deep Q-Network during the training phase.
```

File: deep-rl-dqn.md [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+<img src="assets/78_deep_rl_dqn/qtarget-2.jpg" alt="Q-target"/>
+<img src="assets/78_deep_rl_dqn/qtarget-3.jpg" alt="Q-target"/>
+This leads to a bizarre path of chasing (a significant oscillation in training).
+<img src="assets/78_deep_rl_dqn/qtarget-4.jpg" alt="Q-target"/>
+
+Instead, what we see in the pseudo-code is that we:
+<img src="assets/78_deep_rl_dqn/fixed-q-target-pseudocode.jpg" alt="Fixed Q-target Pseudocode"/>
+  
+- Use a separate network with a fixed parameter for estimating the TD Target
+- At every C step, we copy the parameters from our DQN network to update the target network.
```
Comment (echarlaix) [inline deep-rl-dqn.md:209]: Copy the parameters from our DQN network at every C step to update the target network

File: deep-rl-dqn.md [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+
+Take time to really grasp the material before continuing.
+
+And since the best way to learn and avoid the illusion of competence is **to test yourself**. We wrote a quiz to help you find where **you need to reinforce your study**. Check your knowledge here 👉 **[https://github.com/huggingface/deep-rl-class/blob/main/unit3/quiz.md](https://github.com/huggingface/deep-rl-class/blob/main/unit3/quiz.md)**
+
+Don't hesitate to train your agent in other environments (Pong, Seaquest, QBert, Ms Pac Man). The **best way to learn is to try things on your own!**
+
+We published additional readings in the syllabus if you want to go deeper 👉 **[https://github.com/huggingface/deep-rl-class/blob/main/unit3/README.md](https://github.com/huggingface/deep-rl-class/blob/main/unit3/README.md)**
+
+In the next unit, we’re going to learn about Policy Gradients methods
```
Comment (echarlaix) [inline deep-rl-dqn.md:256]:
```suggestion
In the next unit, we’re going to learn about Policy Gradients methods
```
```suggestion
In the next unit, we’re going to learn about Policy Gradients methods.
```

**Review by echarlaix — APPROVED** [review]
Great work @simoninithomas :fire: !! I added some minor comments

File: deep-rl-dqn.md [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+
+## From Q-Learning to Deep Q-Learning
+
+We learned that **Q-Learning is an algorithm we use to train our Q-Function**, an **action-value function** that determines the value of being at a particular state and taking a specific action at that state.
+
+<figure class="image table text-center m-0 w-full"> <img src="assets/73_deep_rl_q_part2/Q-function.jpg" alt="Q-function"/> <figcaption>Given a state and action, our Q Function outputs a state-action value (also called Q-value)</figcaption> </figure>
+
+The **Q comes from "the Quality" of that action at that state.**
+
+Internally, our Q-function has **a Q-table, a table where each cell corresponds to a state-action value pair value.** Think of this Q-table as **the memory or cheat sheet of our Q-function.**
```
[the remark this answers is not in the document]
Comment (simoninithomas) [reply deep-rl-dqn.md:64]: Error from my side updated ✔️

**Comment by osanseviero** [comment]
You'll also need to update `_blog.yml` :)

**Pull request merged**